streamer/controller_node.py

```diff
@@ -157,29 +150,55 @@ def start(self, output_dir: str,
     self._input_config = InputConfig(input_config_dict)
     self._pipeline_config = PipelineConfig(pipeline_config_dict)
 
+    if not streamer.util.is_url(output_location):
+      # Check if the directory for outputted Packager files exists, and if it
+      # does, delete it and remake a new one.
+      if os.path.exists(output_location):
+        shutil.rmtree(output_location)
+      os.mkdir(output_location)
+    else :
+      # Check some restrictions and other details on HTTP output.
+      if not self._pipeline_config.segment_per_file:
+        raise RuntimeError(
+            'For HTTP PUT uploads, the pipeline segment_per_file setting ' +
+            'must be set to True!')
+
+      if bucket_url:
+        raise RuntimeError(
+            'Cloud bucket upload is incompatible with HTTP PUT support.')
```

Comment:
You have my permission to kill CloudNode once we have a built-in auth proxy that can handle GCS.  We're currently using CloudNode internally for this stream:

https://shaka-player-demo.appspot.com/demo/#asset=https://storage.googleapis.com/shaka-live-assets/player-source.mpd;build=compiled

Reply:
Woohoo! I look forward to the day when we can kill it. :D